src/qibolab/pulses/pulse.py

```diff
@@ -119,6 +119,12 @@ def __hash__(self):
         )
 
 
+class FluxPulse(Pulse):
```

Comment:
Since this is re-introduced, I guess `Pulse.flux` is not needed anymore?

Also, introducing this specialization is a bit against the idea of dropping `pulse.type` after the channel refactoring. In principle, `PulseSequence` should know that a `Pulse` is a `FluxPulse` if it is playing on a flux channel. Generally speaking, it doesn't even need to know that it is a `FluxPulse`, since this is like any other `Pulse` with a shape and a frequency (which happens to be 0), unlike the `Delay` and `VirtualZ` which are very different instructions.

Reply:
If this is needed for the runcard deserialization, I would be inclined to just put `"frequency": 0` in all flux pulses in all runcards. It will be very repetitive, but it is more symmetric from the code point of view. But I have no strong opinion.

Reply:
> Since this is re-introduced, I guess `Pulse.flux` is not needed anymore?

This was never required, just a more convenient constructor in the relevant cases.

> If this is needed for the runcard deserialization, I would be inclined to just put `"frequency": 0` in all flux pulses in all runcards. It will be very repetitive, but it is more symmetric from the code point of view. But I have no strong opinion.

Indeed, it was a compromise. I don't like it too much myself, but it was required to deserialize objects which were missing the `.frequency` attribute.

If you wish, it is just a temporary compatibility layer: `frequency` is missing in the runcard, then it is deserialized as `FluxPulse` (that is nothing else than a normal `Pulse` with some defaults). The idea was never to use the `FluxPulse` object directly.

If you want to drop it, it would only be a good thing, but you have to go through the runcards and add the missing properties (this should be automatically done by deserializing them, with the current `FluxPulse` class, and then serialize them again).